### PR TITLE
Enable WASM AOT option in build and update MonoAOTCompiler

### PR DESCRIPTION
**Pull request by akoeplinger**
Allows creating AOT images for WASM with the AOT compiler and consuming them in the runtime.

Use `make build AOT=1` in `src/mono/netcore/sample/wasm/console` to try it out.

### CODE REVIEW

File: src/mono/mono.proj [hunk 1]
```diff
@@ -186,7 +186,7 @@
       <_MonoMinimal Condition="'$(Configuration)' == 'Release'">,assert_messages</_MonoMinimal>
     </PropertyGroup>
     <ItemGroup Condition="'$(TargetsBrowser)' == 'true'">
-      <_MonoCMakeArgs Include="-DENABLE_MINIMAL=ssa,com,jit,reflection_emit_save,portability,assembly_remapping,attach,verifier,appdomains,shadowcopy,security,sgen_marksweep_conc,sgen_split_nursery,sgen_gc_bridge,sgen_toggleref,logging,remoting,shared_perfcounters,sgen_debug_helpers,sgen_binary_protocol,soft_debug,interpreter,cleanup,mdb,gac,threads,eventpipe,aot,interpreter,qcalls$(_MonoMinimal)"/>
+      <_MonoCMakeArgs Include="-DENABLE_MINIMAL=ssa,com,jit,reflection_emit_save,portability,assembly_remapping,attach,verifier,appdomains,shadowcopy,security,sgen_marksweep_conc,sgen_split_nursery,sgen_gc_bridge,sgen_toggleref,logging,remoting,shared_perfcounters,sgen_debug_helpers,sgen_binary_protocol,soft_debug,interpreter,cleanup,mdb,gac,threads,eventpipe,qcalls$(_MonoMinimal)"/>
```
Comment (vargaz) [inline src/mono/mono.proj:189]: The 'interpreter' option will disable the interpreter as part of the runtime, and compile it into a separate lib using the -DENABLE_INTERP_LIB=1 line.

File: src/mono/netcore/sample/Android/Makefile [hunk 2]
```diff
@@ -2,10 +2,7 @@ MONO_CONFIG=Release
 MONO_ARCH=x64
 DOTNET := ../../../../.././dotnet.sh
 
-all: runtimepack run
-
-runtimepack:
-	../../../../.././build.sh Mono+Libs -os Android -arch $(MONO_ARCH) -c $(MONO_CONFIG)
+all: run
```
Comment (vargaz) [inline src/mono/netcore/sample/Android/Makefile:5]: This was a helper target used during development.

File: src/mono/netcore/sample/iOS/Makefile [hunk 3]
```diff
@@ -10,9 +10,6 @@ appbuilder:
 	$(DOTNET) build -c Release $(TOOLS_DIR)/AotCompilerTask/MonoAOTCompiler.csproj
 	$(DOTNET) build -c Release $(TOOLS_DIR)/AppleAppBuilder/AppleAppBuilder.csproj
 
-runtimepack:
-	../../../../.././build.sh Mono+Libs -os iOS -arch $(MONO_ARCH) -c $(MONO_CONFIG)
-
```
Comment (vargaz) [inline src/mono/netcore/sample/iOS/Makefile:15]: Ditto.

File: src/mono/CMakeLists.txt [hunk 4]
```diff
@@ -290,6 +290,9 @@ elseif(TARGET_SYSTEM_NAME STREQUAL "Emscripten")
   if (CMAKE_BUILD_TYPE STREQUAL "Release")
     add_compile_options(-Os)
   endif()
+  if(NOT HOST_WASM)
+    set(GC_SUSPEND "default")
+  endif()
```
Comment (vargaz) [inline src/mono/CMakeLists.txt:295]: Why is this needed ? It should only be hit for the cross compiler.

File: src/mono/mono.proj (hunk 1 above)
Comment (akoeplinger) [reply src/mono/mono.proj:189]: it's still there, I just removed the duplicated instance of `interpreter` in the line

File: src/mono/netcore/sample/Android/Makefile (hunk 2 above)
Comment (akoeplinger) [reply src/mono/netcore/sample/Android/Makefile:5]: it's building quite a lot of stuff, I thought everyone is using the `make runtime` target in src/mono/wasm/Makefile now :)

if you think it's useful I can bring this back and call that target instead.

File: src/mono/CMakeLists.txt (hunk 4 above)
Comment (akoeplinger) [reply src/mono/CMakeLists.txt:295]: yes this is for the cross compiler, I was hitting some issue otherwise.

Comment (vargaz) [reply src/mono/CMakeLists.txt:295]: Wasn't this fixed by:
https://github.com/dotnet/runtime/pull/44093
?

Comment (akoeplinger) [reply src/mono/CMakeLists.txt:295]: ah yeah, I'll check.

Comment (akoeplinger) [reply src/mono/CMakeLists.txt]: removed it.

**Review by steveisok — APPROVED** [review]

**Review by vargaz — APPROVED** [review]

**Comment by akoeplinger** [comment]
Test failure is due to https://github.com/dotnet/runtime/issues/44626.
